oauth-common/src/main/java/io/strimzi/kafka/oauth/common/HttpUtil.java

```diff
@@ -142,6 +163,32 @@ public static void delete(URI uri, SSLSocketFactory socketFactory, HostnameVerif
         return handleResponse(con, method, uri, responseType);
     }
 
+    private static void configureTLS(HttpURLConnection con, URI uri, SSLSocketFactory socketFactory, HostnameVerifier hostnameVerifier) {
+        if (con instanceof HttpsURLConnection) {
+            HttpsURLConnection scon = (HttpsURLConnection) con;
+            if (socketFactory != null) {
+                scon.setSSLSocketFactory(socketFactory);
+            }
+            if (hostnameVerifier != null) {
+                scon.setHostnameVerifier(hostnameVerifier);
+            }
+        } else if (socketFactory != null) {
+            log.warn("SSL socket factory set but url scheme not https ({})", uri);
+        }
+    }
+
+    private static void configureTimeouts(HttpURLConnection con, int connectTimeout, int readTimeout) {
+        if (connectTimeout <= 0) {
+            throw new IllegalArgumentException("connectTimeout <= 0");
+        }
+        con.setConnectTimeout(connectTimeout * 1000);
+
+        if (readTimeout <= 0) {
+            throw new IllegalArgumentException("readTimeout <= 0");
+        }
+        con.setReadTimeout(readTimeout * 1000);
```

Comment:
Should the options be in milliseconds if milliseconds are used here?

Reply:
Not sure anyone would want to be that precise about it. For one, having the minimum timeout of 1s prevents clients from flooding the authorization server with requests (for example by setting read timeout to 1ms). Primarily the idea is to be able to set the timeout and retry a request if nothing happens for 30 seconds for example. It makes no difference if it is 30s or 30.149s. I don't see why someone would want to set the connect timeout or read timeout to less than 1 second for example. But maybe there is a use case, and I just don't see it.

Reply:
Also, maybe 60 seconds is not the best default for connect timeout and read timeout. Maybe 30 seconds makes more sense? Or maybe read timeout is better to be different than the connect timeout?

Primarily the idea is to not have infinite timeout as that may result in a `hijacked` request thread. It's when network layer has issues, or the server is not responding. The question is when did we wait enough, and when is it time to give up on the request. Exponential backoff is implemented separately from that, but can't work properly if its thread is `hijacked`.

Reply:
I think it is common to us milliseconds as the unit even through you usually set it only to seconds. But if we already have other options in seconds here I guess we should stuck with it.

Reply:
> Also, maybe 60 seconds is not the best default for connect timeout and read timeout. Maybe 30 seconds makes more sense? Or maybe read timeout is better to be different than the connect timeout?

> Primarily the idea is to not have infinite timeout as that may result in a hijacked request thread. It's when network layer has issues, or the server is not responding. The question is when did we wait enough, and when is it time to give up on the request. Exponential backoff is implemented separately from that, but can't work properly if its thread is hijacked.

I think these are good questions. But not sure I have the answers I'm afraid.